docs/reference/snapshot-restore/repository-s3.asciidoc

```diff
@@ -541,13 +541,17 @@ MinIO-backed repositories as well as repositories stored on AWS S3. Other
 S3-compatible storage systems may also work with {es}, but these are not
 covered by the {es} test suite.
 
-Note that some storage systems claim to be S3-compatible but do not faithfully
-emulate S3's behaviour in full. The `repository-s3` type requires full
-compatibility with S3. In particular it must support the same set of API
-endpoints, return the same errors in case of failures, and offer consistency and
-performance at least as good as S3 even when accessed concurrently by multiple
-nodes. You will need to work with the supplier of your storage system to address
-any incompatibilities you encounter. Please do not report {es} issues involving
+There are many systems, including some from very well-known storage vendors,
+which claim to offer an S3-compatible API despite failing to emulate S3's
+behaviour in full. If you are using such a system for your snapshots, consider
+using a <<snapshots-filesystem-repository,shared filesystem repository>> based
+on a standardized protocol such as NFS to access your storage system instead.
```

Comment:
Do we want to have a bit softer statement to say something like "If you are using such a system for snapshots, especially if such system does not pass verify snapshot repository tests, consider ..."? Is it helpful at all? Or maybe it is more misleading since the verify API may adapot new S3 features and what was compatible may become incompatible later?

Reply:
I think the "consider" is soft enough. If the system doesn't pass verification then it absolutely shouldn't be used, but yeah we've seen enough systems that pass today but then fail some stricter checks in a future version. I think NFS is always worth considering in these situations, it just doesn't have the same problems.